enhancements/opm-content-management.md

```diff
@@ -0,0 +1,145 @@
+---
+title: opm-content-management 
+authors:
+  - "@anik120"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# OPM Content Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This enhancement is a proposal for extending the existing capabilites of the `opm` command line utility, to provide more fine grained control over the management of catalog indexes.   
+
+`opm` will have significant improvement to UX for editing an index (removing individual bundles from a package, removing channels from a package). Indexes can also be queried directly for inspection, and update graph visualization for individual packages will be added. Bundle validation without a deamon will is also proposed in this enhancement. 
+
+## Motivation
+
+As more and more operators are migrated to the new bundle format, and more index images are created for Operator Catalogs, creators of catalogs of operators, such as the Red Hat pipeline responsible for maintaining the `redhat-operators` catalog, should have more fine grained control over the management of the catalogs using the `opm` tooling.  
+
+### Goals
+
+- Allow `opm` users to perform bundle validations in environments without a Docker daemon. Bundle validation also includes validation of bundle image labels against `metadata/annotations.yaml`
+- Allow `opm` users to safely edit indexes, which includes removing bundle version from a package in an index, and removing a channel from a bundle. 
+- Allow `opm` users to inspect catalog content and return the update graph of Operators in machine-readable format.
+
+### Non-Goals
+
+
+## Proposal
+ 
+### Implementation Details
+
+### Validate the bundle image labels against bundle annotations
+
+Currently, `opm alpha bundle validate` validates only the content of `metadata/annotations.yaml` file, but not the image labels of the DockerFile. The image labels are used when the bundle is added to an index image. As a result, if there are mismatches between the content of the bundle image (i.e `metadata/annotations.yaml`) and the bundle image labels, the bundle image is invalid and could potentially cause issues when it is added to an index image.
+
+The `opm alpha bundle validate` command will be enhanced with an extra step that'll check to see if the contents of `metadata/annotations.yaml` matches with the labels of the bundle image. 
```

Comment:
```suggestion
The `opm alpha bundle validate` command will be enhanced with an extra step that will check to see if the contents of `metadata/annotations.yaml` matches with the labels of the bundle image. 
```

Reply:
what does it mean to match? Is it ok to have labels that are not in the annotations? annotations that are not in the labels?

Reply:
@gallettilance the idea is to have the exact number of annotations as the number of labels. I'll be more specific about this in this para.